--- gtk+-2.99.0/configure.ac.wiget	2011-01-06 04:55:33.000000000 +0100
+++ gtk+-2.99.0/configure.ac	2011-01-09 21:32:51.429526702 +0100
@@ -324,7 +324,7 @@ else
 fi
 
 # strip leading space
-GDK_BACKENDS=${GDK_BACKENDS/# }
+GDK_BACKENDS=${GDK_BACKENDS# }
 
 AC_SUBST(GDK_BACKENDS)
 
@@ -1162,7 +1162,7 @@ if test "x$enable_x11_backend" == xyes;
   AM_CONDITIONAL(USE_X11, true)
 
   # strip leading space
-  X_EXTENSIONS=${X_EXTENSIONS/# }
+  X_EXTENSIONS=${X_EXTENSIONS# }
 
 else
   XPACKAGES=
@@ -1658,7 +1658,7 @@ perf/Makefile
 AC_OUTPUT
 
 # beautify the immodule list a bit
-included_immodules=${included_immodules//,/ }
+included_immodules="$(echo $included_immodules|sed -e 's/,/ /g')"
 included_immodules=${included_immodules:-none}
 
 echo "configuration:"
