--- gtk+-3.24.36/meson.build.orig	2022-12-24 22:25:03.301902858 +0100
+++ gtk+-3.24.36/meson.build	2022-12-24 22:26:43.555542218 +0100
@@ -914,7 +914,7 @@
 
 pkgconf.set('GDK_PRIVATE_PACKAGES',
             ' '.join(gio_packages + x11_pkgs + wayland_pkgs + cairo_backends +
-                     ['epoxy', epoxy_req] + cloudproviders_packages +
+                     ['epoxy', epoxy_req] +
                      ['fribidi', fribidi_req]))
 
 gtk_packages = ' '.join([
@@ -933,6 +933,7 @@
   gtk_private_packages += ['pangoft2']
 endif
 gtk_private_packages += gio_packages
+gtk_private_packages += cloudproviders_packages
 
 pkgconf.set('GTK_PRIVATE_PACKAGES', ' '.join(gtk_private_packages))
 
